docs/public/specifications/tests/Connectivity-TP/connectivity04.md

```diff
@@ -0,0 +1,235 @@
+CN04_IPV6_DIFFERENT_PREFIX  | INFO    | ns_ip_list                  | The following name server(s) are announced in unique IPv6 prefix(es): "{ns_ip_list}"
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine Profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [Argument List].
+
+## Test procedure
+
+1. Create the following empty sets:
+   1. Name server IP address ("NS IPv4 IPs")
+   2. Name server IP address ("NS IPv6 IPs")
+   3. IP prefix and name server IP address ("IPv4 Prefix")
+   4. IP prefix and name server IP address ("IPv6 Prefix")
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+3. For each name server IP in *Name Server IP* do:
+   1. Add IPv4 addresses to the *NS IPv4 IPs* set.
+   2. Add IPv6 addresses to the *NS IPv6 IPs* set.
+
+4. For each IP address in *NS IPv4 IPs* and *NS IPv6 IPs* ("NS IP Address"),
+   respectively, do:
+   1. Determine the IP prefix in which *NS IP Address* is announced
+      using *Prefix Database* (details in sections below).
+   2. Add found IP prefix, if any, with *NS IP Address* to the *IPv4 Prefix*
+      and *IPv6 Prefix* sets, respectively.
+
+5. If the *IPv4 Prefix* set is non-empty, then do:
+   1. For each IP prefix in the set that has two or more members, output
+      *[CN04_IPV4_SAME_PREFIX]* with the prefix and list of all members for that prefix.
+   2. For each IP prefix in the set that has exactly one member, output
+      *[CN04_IPV4_DIFFERENT_PREFIX]* with the combined set of their associated members.
+
+6. If the *IPv6 Prefix* set is non-empty, then do:
+   1. For each IP prefix in the set that has two or more members, output
+      *[CN04_IPV6_SAME_PREFIX]* with the prefix and list of all members for that prefix.
+   2. For each IP prefix in the set that has exactly one member, output
+      *[CN04_IPV6_DIFFERENT_PREFIX]* with the combined set of their associated members.
+
+## Outcome(s)
+
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *[ERROR]* or *[CRITICAL]*.
+
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *[WARNING]*, but no message with severity level
+*ERROR* or *CRITICAL*.
+
+In other cases, no message or only messages with severity level
+*[INFO]* or *[NOTICE]*, the outcome of this Test Case is "pass".
+
+## Special procedural requirements
+
+This Test Case is dependent on one of two possible services that can provide
+ASN lookup (Cymru or RIPE RIS). The service must be available over the network.
+
+The *Child Zone* must be a valid name meeting
+"[Requirements and normalization of domain names in input]".
+
+### Cymru prefix lookup
+
```

Comment:
Should this section be a sub-section to "Special procedural requirements"? Maybe a header
```
## Prefix lookup methods

Use the prefix method set in *Prefix Database* in "Inputs".

```

Reply:
Don’t forget to update the table of contents too. 🙂